### ISSUE TITLE
Use Android 30 CloseGuard

### ISSUE THREAD

**Issue by yschimke** [square/okhttp#6748]
```
2021-07-06 15:42:39.925 6517-6517/xxx E/STRICTMODE: android.os.strictmode.NonSdkApiUsedViolation: Ldalvik/system/CloseGuard;->get()Ldalvik/system/CloseGuard;
        at android.os.StrictMode.lambda$static$1(StrictMode.java:416)
        at android.os.-$$Lambda$StrictMode$lu9ekkHJ2HMz0jd3F8K8MnhenxQ.accept(Unknown Source:2)
        at java.lang.Class.getDeclaredMethodInternal(Native Method)
        at java.lang.Class.getPublicMethodRecursive(Class.java:2079)
        at java.lang.Class.getMethod(Class.java:2066)
        at java.lang.Class.getMethod(Class.java:1693)
        at okhttp3.internal.platform.AndroidPlatform$CloseGuard.get(AndroidPlatform.java:379)
        at okhttp3.internal.platform.AndroidPlatform.<init>(AndroidPlatform.java:58)
        at okhttp3.internal.platform.Android10Platform.<init>(Android10Platform.java:32)
        at okhttp3.internal.platform.Android10Platform.buildIfSupported(Android10Platform.java:82)
```

https://developer.android.com/reference/android/util/CloseGuard

**Comment by yschimke** [comment]
I have enough questions about our usage, no plans to rush this in https://github.com/square/okhttp/pull/6750

**Comment by yschimke** [comment]
@swankjesse What are your thoughts on backporting this to 3.12.x?

Seems like we should based on support dates

> The OkHttp 3.12.x branch supports Android 2.3+ (API level 9+) and Java 7+. These platforms lack support for TLS 1.2 and should not be used. But because upgrading is difficult, we will backport critical fixes to the 3.12.x branch through December 31, 2021.
